nucypher/network/middleware.py

```diff
@@ -102,11 +115,22 @@ def parse_node_or_host_and_port(self, node, host, port):
             raise ValueError("You need to pass either the node or a host and port.")
         return host, port, self.library
 
+    def _determine_timeout(self, provided_timeout) -> Union[float, Tuple[float, float]]:
+        # provided timeout (could be None) is really for reading not initial connection timeout,
+        # but use as fallback for connect timeout
```

Comment:
I didn't get that. How are these two used? Is `connect_timeout` for the TCP connection or finding peers? Is `read_timeout` for uhh reading the data from said request?

Reply:
Yes. Basically there are two timeouts for the `requests` library:
- `connect timeout`: number of seconds Requests will wait for your client to establish a connection to a remote machine
- `read timeout`: number of seconds the client will wait for the server to send a response

If you only specify one value - then the same value is used for both. When someone provides a timeout via the method call, they are really intending to specify the `read timeout`, so we use our own internal connect timeout of 3.05 seconds, which is sufficient for connection to nodes. 

The read timeout can be unnecessarily large if we use it for the connect timeout.

Modified the code a little to attempt to make that clearer.

(see https://requests.readthedocs.io/en/latest/user/advanced/#timeouts)